module_analysis/models/ir_module_type_rule.py

```diff
@@ -0,0 +1,29 @@
+# Copyright (C) 2019-Today: GRAP (<http://www.grap.coop/>)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class IrModuleType(models.Model):
+    _name = "ir.module.type.rule"
+    _description = "Modules Types Rules"
+    _order = "sequence"
+
+    sequence = fields.Integer()
+
+    module_domain = fields.Char(required=True, default="[]")
```

Comment:
```suggestion
    module_domain = fields.Binary(required=True, default=[])
```
I'm not sure if this would require a migration script.

Reply:
@rrebollo It’s not possible to switch to a binary field because the domain cannot be applied at the view level.